modules/javafx.controls/src/main/java/javafx/scene/control/DialogPane.java

```diff
@@ -1057,7 +1059,7 @@ private void updateButtons(ButtonBar buttonBar) {
 
         boolean hasDefault = false;
         for (ButtonType cmd : getButtonTypes()) {
-            Node button = buttonNodes.computeIfAbsent(cmd, dialogButton -> createButton(cmd));
+            Node button = buttonNodes.get(cmd);
```

Comment:
This change is not needed for the fix. 

I removed it, because #computeIfAbsent() isn't needed anymore.
The (now) first listener will run and create/remove buttons based off the changes to #getButtonTypes().
The second listener (which is calling this method) will use all created buttons (to put them inside the ButtonBar).

-> So only a simple #get() is needed.
#computeIfAbsent() basically masked this bug before, as it was also creating a button, if missing. But as mentionend, when we are here, all buttons should be already created by the first listener.